spec/requests/api/v3/providers/suggest_provider_spec.rb

```diff
@@ -0,0 +1,90 @@
+require "rails_helper"
+
+describe "GET /provider-suggestions" do
+  let(:jsonapi_renderer) { JSONAPI::Serializable::Renderer.new }
+  let(:organisation) { create(:organisation) }
+  let(:provider1) { create(:provider, provider_name: "PROVIDER 1", organisations: [organisation]) }
+  let(:provider2) { create(:provider, provider_name: "PROVIDER 2", organisations: [organisation]) }
+  let(:next_recruitment_cycle) { find_or_create(:recruitment_cycle, :next) }
+  let(:provider3) { create(:provider, provider_name: provider2.provider_name, organisations: [organisation], recruitment_cycle: next_recruitment_cycle) }
+  let(:provider4) { create(:provider, organisations: [organisation], recruitment_cycle: next_recruitment_cycle) }
+
+  before do
+    provider1
+    provider2
+    provider3
+    provider4
+  end
+
+  it "searches for a particular provider for the user in the current recruitment cycle" do
+    get "/api/v3/provider-suggestions?query=#{provider2.provider_name}"
```

Comment:
This could be in a context and then only needs `provider2`. You could argue that it needs another one that isn't returned to make it a better test I suppose.

Reply:
I would agree that we need another provider in there to demonstrate that it doesn't get returned. It's cases like this that was driving my proposal for standard test data setup.

Reply:
👍